Dockerfile

```diff
@@ -1,7 +1,23 @@
-FROM golang:latest
+# builder
+FROM golang:latest AS builder
 RUN mkdir /app
 ADD . /app/
 WORKDIR /app
+
 RUN go mod download
-RUN go build -o main .
-CMD ["/app/main", "--portHTTP=80"]
\ No newline at end of file
+RUN make build
+RUN ls -l .
+
+# certificates
+FROM alpine:latest as certs
+RUN apk --no-cache add ca-certificates
+
+# runner
+FROM scratch
+#WORKDIR /app
+
+COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
```

Comment:
Always a pleasure to explain!

This is not required right now. But I did it, in case of future features.
The goal here is to have the smallest container possible, as only the binary file is required. When you use a `scratch` container, it's an empty one: **0Kib**.

But, when you want to do an API call from the Golang app, you mostly need to do HTTPS calls. And in order to do that, you need to have the certificate authority to validate the SSL certificates. Without that file, no HTTPS calls are possible.

Here, I simply use the one from the `alpine:latest` to put the file in the scratch container.

This is a common best practice in Golang containers.

Hope that helps :)

TLDR:
 - use stage container to get the CA file
 - copy it to the scratch container beside the binary

Reply:
Would be great if you could put the short version of that comment into the docker file itself. Even though it might seem verbose or useless, it will surely help future readers of the code.